packages/in_app_purchase/lib/src/in_app_purchase/product_details.dart

```diff
@@ -58,4 +79,9 @@ class ProductDetailsResponse {
   /// There's multiple platform specific reasons that product information could fail to be fetched,
   /// ranging from products not being correctly configured in the storefront to the queried IDs not existing.
   final List<String> notFoundIDs;
+
+  /// The error occurred during the request.
```

Comment:
I think it's important here to say that this is an exception that happened somewhere in the platform layer. It's possible to have this be null but still have all invalid IDs, which is confusing. :|

Something like:

> A caught platform exception thrown while querying the purchases.
>
> It's possible for this to be null but for there still to be `notFoundIds` in cases where the request itself was a success but the requested IDs could not be found.

It may be better for consistency's sake to populate this with something like a 404 if we detect `notFoundIDs` here, but it would change how we're handling the object in other places. WDYT?

Reply:
I think it is ok since we also documented the notFoundIDs saying it can have different reasons. The comment you suggested is a good addition to this doc and should be sufficient.